packages/visual-editor/src/components/migrations/README.md

```diff
@@ -0,0 +1,75 @@
+# Migrations
+
+Migrations ensure backwards compatibility for component updates. When a site is updated, it immediately begins
+using the new components however, the published and save state data is old. This data is run through
+`migrate` prior to being rendered so that any migrations will be applied and the data will match with the components.
+
+## When do I need to add a migrations?
+
+1. A component is renamed.
+
+Without a migration, old usages of the component will disappear from the live page and the user will be
+required to add the component to the layout again.
+
+2. Component props are updated
+
+Without a migration, new props will not have any data and data contained in old props will not be used.
+
+3. Component fields are updated
+
+If you update some fields, such as select, the new options may not match the old data.
+
+4. A component is removed (optional)
+
+A removed component will automatically disappear from the live page, however adding a
```

Comment:
A removed component will automatically disappear from the live page as well as showing a "No Configuration for" message in the editor. Add a removed migration to handle this.